server/sonar-web/src/main/js/apps/users/components/TokensForm.tsx

```diff
@@ -0,0 +1,202 @@
+      tokens => {
+        if (this.mounted) {
+          this.setState({ loading: false, tokens });
+        }
+      },
+      () => {
+        if (this.mounted) {
+          this.setState({ loading: false });
+        }
+      }
+    );
+  };
+
+  handleCloseClick = (evt: React.SyntheticEvent<HTMLAnchorElement>) => {
+    evt.preventDefault();
+    this.handleClose();
+  };
+
+  handleClose = () => {
+    if (this.state.hasChanged) {
+      this.props.onUpdateUsers();
+    }
+    this.props.onClose();
+  };
+
+  handleGenerateToken = (evt: React.SyntheticEvent<HTMLFormElement>) => {
+    evt.preventDefault();
+    if (this.state.newTokenName.length > 0) {
+      this.setState({ generating: true });
+      generateToken({ name: this.state.newTokenName, login: this.props.user.login }).then(
+        newToken => {
+          if (this.mounted) {
+            this.fetchTokens();
+            this.setState({ generating: false, hasChanged: true, newToken, newTokenName: '' });
+          }
+        },
+        () => {
+          if (this.mounted) {
+            this.setState({ generating: false });
+          }
+        }
+      );
+    }
+  };
+
+  handleRevokeToken = () => {
+    this.setState({ hasChanged: true });
+    this.fetchTokens();
+  };
+
+  handleNewTokenChange = (evt: React.SyntheticEvent<HTMLInputElement>) =>
+    this.setState({ newTokenName: evt.currentTarget.value });
+
+  renderItems() {
+    const { tokens } = this.state;
+    if (tokens.length <= 0) {
+      return (
+        <tr>
+          <td colSpan={3} className="note">
+            {translate('users.no_tokens')}
+          </td>
+        </tr>
+      );
+    }
+    return tokens.map(token => (
+      <TokensFormItem
+        key={token.name}
+        token={token}
+        onRevokeToken={this.handleRevokeToken}
+        user={this.props.user}
+      />
+    ));
+  }
+
+  render() {
+    const { generating, loading, newToken, newTokenName, tokens } = this.state;
+    const header = translate('users.tokens');
+    const customSpinner = (
+      <tr>
+        <td>
+          <i className="spinner" />
+        </td>
+      </tr>
+    );
+    return (
+      <Modal contentLabel={header} onRequestClose={this.handleClose}>
+        <header className="modal-head">
+          <h2>{header}</h2>
+        </header>
+        <div className="modal-body modal-container">
+          <h3 className="spacer-bottom">{translate('users.generate_tokens')}</h3>
+          <form id="generate-token-form" onSubmit={this.handleGenerateToken} autoComplete="off">
+            <input
+              className="spacer-right"
+              type="text"
+              maxLength={100}
+              onChange={this.handleNewTokenChange}
+              placeholder={translate('users.enter_token_name')}
+              required={true}
+              value={newTokenName}
+            />
+            <button
+              className="js-generate-token"
+              disabled={generating || newTokenName.length <= 0}
```

Comment:
be consistent with checking the length: sometimes it's `<= 0`, sometimes it's `=== 0`